File: cse-analyzer/backend/app/routers/watchlist.py
from fastapi import APIRouter, Depends
from app.middleware.auth_middleware import get_current_user
from app.services import watchlist_service

router = APIRouter(prefix="/watchlist", tags=["watchlist"])


@router.get("")
async def get_watchlist(current_user: dict = Depends(get_current_user)):
    return await watchlist_service.get_watchlist(current_user["user_id"])


@router.post("/{symbol}")
async def add_symbol(symbol: str, current_user: dict = Depends(get_current_user)):
    return await watchlist_service.add_to_watchlist(current_user["user_id"], symbol)


@router.delete("/{symbol}")
async def remove_symbol(symbol: str, current_user: dict = Depends(get_current_user)):
    return await watchlist_service.remove_from_watchlist(current_user["user_id"], symbol)


@router.get("/check/{symbol}")
async def check_watching(symbol: str, current_user: dict = Depends(get_current_user)):
    watching = await watchlist_service.is_watching(current_user["user_id"], symbol)
    return {"watching": watching}

File: cse-analyzer/frontend/src/api/watchlist.ts
import client from "./client";

export interface WatchlistItem {
  symbol: string;
  added_at: string;
  name?: string;
  sector?: string;
  close?: number;
  pct_change?: number;
  volume?: number;
  market_cap?: number;
  date?: string;
}

export const getWatchlist    = () => client.get<WatchlistItem[]>("/watchlist").then(r => r.data);
export const addToWatchlist  = (symbol: string) => client.post(`/watchlist/${symbol}`).then(r => r.data);
export const removeFromWatchlist = (symbol: string) => client.delete(`/watchlist/${symbol}`).then(r => r.data);
export const checkWatching   = (symbol: string) => client.get<{ watching: boolean }>(`/watchlist/check/${symbol}`).then(r => r.data);
